test/nodes/core/core/80-template_spec.js

```diff
@@ -37,9 +37,40 @@ describe('template node', function() {
             n2.on("input", function(msg) {
                 msg.should.have.property('topic', 'bar');
                 msg.should.have.property('payload', 'payload=foo');
+                msg.should.have.property('template', '{{payload}}');
                 done();
             });
-            n1.receive({payload:"foo",topic: "bar"});
+            n1.receive({payload:"foo",topic: "bar", template: "{{payload}}"});
+        });
+    });
+
+    it('should modify template from msg.template', function(done) {
+        var flow = [{id:"n1", type:"template", field:"template", template:"",wires:[["n2"]]},{id:"n2",type:"helper"}];
+        helper.load(templateNode, flow, function() {
+            var n1 = helper.getNode("n1");
+            var n2 = helper.getNode("n2");
+            n2.on("input", function(msg) {
+                msg.should.have.property('topic', 'bar');
+                msg.should.have.property('payload', 'foo');
+                msg.should.have.property('template', 'payload=foo');
+                done();
+            });
+            n1.receive({payload:"foo", topic: "bar", template: "payload={{payload}}"});
```

Comment:
template here should be different from one set in node to make sure the one in the node has priority.. (otherwise it will pass test either way)